logsapi/event.go

```diff
@@ -101,6 +99,23 @@ func (lc *Client) ProcessLogs(
 					lc.logger.Warn("report event request id didn't match the previous event id")
 					lc.logger.Debug("Log API runtimeDone event request id didn't match")
 				}
+			case FunctionLog:
+				if metadataContainer != nil && len(metadataContainer.Metadata) > 0 {
```

Comment:
Could we neaten this a little by doing this?

```go
if metadataContainer == nil || len(metadataContainer.Metadata) == 0 {
    lc.logger.Warn("Function log received before metadata populated, quietly dropping log")
    continue
}

lc.logger.Debug("Received function log")
...
```

Reply:
I have moved the nil handling inside `ProcessFunctionLog`, felt it would be a bit cleaner this way. Let me know your thoughts.

Reply:
Looks good, thanks!